torch/csrc/jit/mobile/import.cpp

```diff
@@ -226,6 +252,113 @@ void parseMethods(
+    if (element.isTuple()) {
+      //  The second item of elements is a list of methods, like forward
+      //  method, example:
+      //  ('__torch__.m.forward',
+      //      (('instructions',
+      //          (('STOREN', 1, 2),
+      //              ('DROPR', 1, 0),
+      //              ('MOVE', 2, 0),
+      //              ('OP', 0, 0),
+      //              ('RET', 0, 0))),
+      //          ('operators', (('aten::Int', 'Tensor'),)),
+      //          ('constants', ()),
+      //          ('types', ()),
+      //          ('register_size', 2)
+      //      )
+      //  )
+      const auto& methods = element.toTuple()->elements();
+      for (const auto& method : methods) {
+        if (method.isTuple()) {
+          //  method example:
+          //  (('instructions',
+          //      (('STOREN', 1, 2),
+          //          ('DROPR', 1, 0),
+          //          ('MOVE', 2, 0),
+          //          ('OP', 0, 0),
+          //          ('RET', 0, 0))),
+          //      ('operators', (('aten::Int', 'Tensor'),)),
+          //      ('constants', ()),
+          //      ('types', ()),
+          //      ('register_size', 2)
+          //  )
+          const auto& method_elements = method.toTuple()->elements();
+          for (const auto& method_element : method_elements) {
+            //  method_element example:
+            //  ('instructions',
+            //      (('STOREN', 1, 2),
+            //          ('DROPR', 1, 0),
+            //          ('MOVE', 2, 0),
+            //          ('OP', 0, 0),
+            //          ('RET', 0, 0)),
+            //  )
+            // A list of if condition statement, trying to locate the
+            // 'constants' field.
+            if (method_element.isTuple()) {
+              const auto& key_values_vector =
+                  method_element.toTuple()->elements();
+              if (key_values_vector.size() == 2) {
+                const auto& key = key_values_vector[0];
+                const auto& values = key_values_vector[1];
+                // Find constant fields
+                if (key.isString() && key == constants_ir) {
+                  if (values.isTuple()) {
+                    const auto& constant_values = values.toTuple()->elements();
+                    for (const auto& constant_value : constant_values) {
+                      if (constant_value.isTuple()) {
+                        const auto& constant_value_tuple =
+                            constant_value.toTuple()->elements();
+                        if (constant_value_tuple.size() == 2 &&
+                            constant_value_tuple[0] == tensor_jit_index_ir &&
+                            constant_value_tuple[1].isInt()) {
+                          return true;
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  return false;
```

Comment:
@cccclai  The reason I asked for refactoring this method is because of this string of closing braces! There are 12 levels of indentation here - I'm sure we can reduce that to 2-3 tops by refactoring into separate methods to make it more readable :)